server/src/server.js

```diff
@@ -1091,10 +1098,13 @@ app.get("/api/fxa-oauth/login/*", function(req, res, next) {
     state = `${state}${END_POINT_SEPARATOR}${req.params[0]}`;
     const profile = "profile";
     res.redirect(`${config.fxa.oAuthServer}/authorization?client_id=${encodeURIComponent(config.fxa.clientId)}&redirect_uri=${encodeURIComponent(redirectUri)}&state=${encodeURIComponent(state)}&scope=${encodeURIComponent(profile)}`);
-  }).catch(next);
+  } catch (err) {
+    mozlog.warn("fxa-oauth-login-failed", {err});
+    next(err);
+  }
 });
 
-app.get("/api/fxa-oauth/confirm-login", function(req, res, next) {
+app.get("/api/fxa-oauth/confirm-login", async function(req, res, next) {
   if (!req.deviceId) {
```

Comment:
Wouldn't this `if` stop what we're  trying to do right away?

Reply:
Oh, maybe I'm confusing this particular PR with the overall goal of having FxA work without device id?

Reply:
Yes,  this check will change to include accountId with PR that allows shots access and authentications on non-Firefox browser